packages/ra-ui-materialui/src/form/FormTab.js

```diff
@@ -20,7 +20,7 @@ const hiddenStyle = { display: 'none' };
 
 class FormTab extends Component {
     renderHeader = ({ className, label, icon, value, translate, ...rest }) => {
-        const to = { pathname: value, search: 'skipFormReset' }; // FIXME use location state when https://github.com/supasate/connected-react-router/issues/301 is fixed
```

Comment:
The issue isn't fixed.
Oh wait, is that because you removed the saga watching on the location change to reset the form?